### ISSUE TITLE
Erroneous exchange CO2 intensities

### ISSUE THREAD

**Issue by silimotion** [electricitymaps/electricitymaps-contrib#5924]
## Description
For some time now I've noticed that some exchanges don't have the CO2 intensity expected. Here is an example:
![Screenshot_20230920_202850](https://github.com/electricitymaps/electricitymaps-contrib/assets/84802276/705f3f0e-ddc5-46a0-bfa4-8d69cf51eb58)
This data then reaches the carbon emissions tab and skews the data a bit.
I think I have discovered what is causing this. When data is available in more granular formats (e.g. 5 min in Australia), the backend calculates the value of the exchange intensity by multiplying the exchange amount by the intensity in the exporter zone for each interval, and then adds up all the intervals. It divides by the total electricity exchanged to get the intensity. If there is a change in flow at the middle of the hour interval, the final intensity is a mix of the intensity of both zones. However, when calculating total emissions in the carbon tab (I haven't checked others), the amount of carbon exported is the total electricity exported by the calculated carbon intensity of the electricity exchange. In the Australia example, this means South Australia exports more carbon than should be (They may be importing actually).

I believe this is how the system is working because I have calculated the emissions following this method and got the same value:
![Screenshot_20230920_203458](https://github.com/electricitymaps/electricitymaps-contrib/assets/84802276/7c99b6b7-be21-400b-b266-5e0dde32583a)

((120+78+78+58+39+24+1+26)*59 + (22+6+44+20)*462)/(120+78+78+58+39+24+1+26+22+6+44+20) = 130

I have used 59 and 462 as the respective zones CO2 intensities for the whole hour. I have not calculated them for each 5 minute interval to save time.

So, in summary, the calculated averages are made for the total energy exchange, but don't take into account the direction of exchange.

**Comment by VIKTORVAV99** [comment]
Not really sure what is causing it but I can confirm the issue.
It also looks like the openNEM parser we use could use an update as it now looks like solar is reported every 5 min and that it should use the new parser classes.

This might fix the issue for this exchange in particular as then the data resolution would match. But we should look into this as it looks like an underlying issue.

**Comment by unitrium** [comment]
I also see the issue right now on the Spain->France exchange
<img width="808" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/38283096/d0a3ed9e-4d2a-466f-b5dc-4c62947cc976">
<img width="764" alt="image" src="https://github.com/electricitymaps/electricitymaps-contrib/assets/38283096/7a39ddcd-0da5-4ced-9813-28c32420db9e">

Not entierly sure it's related to the granularity of the data because our backend should take care of that and realign the values. Nevertheless, I will put this issue in our triage queue so that we have a deeper look as soon as possible.
As for the parser update, I can make it @VIKTORVAV99 but I'm not sure it will fix the issue.

**Comment by VIKTORVAV99** [comment]
It's worth a try as the parser is due an update either way. But if it affects other regions it probably won't fix it.

**Comment by madsnedergaard** [comment]
This has been fixed by @tonypls and it was released 2 weeks ago :)
